pydrad/parse/parse.py

```diff
@@ -525,6 +436,102 @@ def grid_edges_right(self) -> u.cm:
         """
         return self.grid_edges[1:]
 
+    @property
+    @u.quantity_input
+    def coordinate(self) -> u.cm:
+        """
+        Coordinate in the field-aligned direction
+        """
+        return self.grid_centers
+
+    @property
+    @u.quantity_input
+    def electron_mass_density(self) -> u.Unit('g cm-3'):
+        # TODO: Account for possible presence of both electron
+        # and ion mass densities. If the electron mass density
+        # is present in this file, it will shift all of the
+        # indices in the .amr file by one.
+        if 'electron_mass_density' in self._amr_data.colnames:
+            return self._amr_data['electron_mass_density']
+        return self.ion_mass_density
+
+    @property
+    @u.quantity_input
+    def ion_mass_density(self) -> u.Unit('g cm-3'):
+        return self._amr_data['ion_mass_density']
+
+    @property
+    @u.quantity_input
+    def momentum_density(self) -> u.Unit('g cm-2 s-1'):
+        return self._amr_data['momentum_density']
+
+    @property
+    @u.quantity_input
+    def electron_energy_density(self) -> u.Unit('erg cm-3'):
+        return self._amr_data['electron_energy_density']
+
+    @property
+    @u.quantity_input
+    def ion_energy_density(self) -> u.Unit('erg cm-3'):
+        return self._amr_data['ion_energy_density']
+
+    @property
+    @u.quantity_input
+    def velocity(self) -> u.cm/u.s:
+        if hasattr(self, '_phy_data'):
+            return self._phy_data['velocity']
+        return self.momentum_density / self.ion_mass_density
+
+    @property
+    @u.quantity_input
+    def ion_density(self) -> u.Unit('cm-3'):
```

Comment:
Should we consider name changes since this is being refactored?  This is technically the _hydrogen_ density, which includes neutrals, and (I think?) does _not_ include the density of ions from other elements.  

Ditto for the other "ion" properties.

Reply:
Ultimately it makes no difference what the variable is named, so long as it's clear what the variable is.  Perhaps just a note in the documentation is sufficient?

Reply:
I've now renamed all of the `ion_` quantities to `hydrogen_` instead. I think I got all of them, but have a look and let me know if I missed any and if you agree with all of the naming conventions.